### PR TITLE
Add Makefile for building for MacOS & Linux👷‍♂️

### PR DESCRIPTION
**Pull request by TGotwig**
relates #606

### CODE REVIEW

**Comment by foresterre** [comment]
Hi, thanks for the PR.

The linux build seems to work perfectly, but I have some issues with the mac version (compiling from Ubuntu):

<details>
<summary>Relevant log</summary>

```
Building for MacOS... 🍏
cross build --release --target=x86_64-apple-darwin
   Compiling autocfg v1.0.1
   Compiling libc v0.2.79
   Compiling cfg-if v0.1.10
   Compiling proc-macro2 v1.0.24
   Compiling lazy_static v1.4.0
   Compiling unicode-xid v0.2.1
   Compiling bitflags v1.2.1
   Compiling syn v1.0.42
   Compiling maybe-uninit v2.0.0
   Compiling crc32fast v1.2.0
   Compiling rayon-core v1.8.1
   Compiling scopeguard v1.1.0
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `lazy_static`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: build failed
make[1]: *** [Makefile:16: build-mac] Error 101
make[1]: Leaving directory '/home/al/ws/sic'
make: *** [Makefile:6: build] Error 2

``` 
</details>

In particular, the `x86_64-apple-darwin` target doesn't seem to be supported by cross ([supported toolchains](https://github.com/rust-embedded/cross#supported-targets)).

Now there may be some ways to cross compile [anyways](https://wapl.es/rust/2019/02/17/rust-cross-compile-linux-to-macos.html) but I haven't looked into that so far...

Another option would be to build releases on the CI instead of manually (which supports MacOS runners). I've been thinking about doing that for a while now, but haven't given it the necessary attention yet. 

What do you think the best option would be?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	make shasum
+
+build-linux:
+	@echo 'Building for Linux... 🐧'
+	cross build --release --target=x86_64-unknown-linux-musl
+	mkdir -p target/release-archives && tar -C target/x86_64-unknown-linux-musl/release -czf target/release-archives/sic-linux.tar.gz sic
+
+build-mac:
+	@echo 'Building for MacOS... 🍏'
+	cross build --release --target=x86_64-apple-darwin
```
Comment (foresterre) [inline Makefile:16]: cross doesn't seem to [support](https://github.com/rust-embedded/cross#supported-targets) this target; any idea whether I am doing something wrong? :slightly_smiling_face:

**Review by foresterre — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (TGotwig) [reply Makefile:16]: Does it work when you delete the whole `target` folder first? 🤔

Comment (foresterre) [reply Makefile:16]: I tried both manually removing the target folder and `cargo clean` which presumably does the same thing; unfortunately I get the same result :cry:

Comment (TGotwig) [reply Makefile:16]: Ah yeah right, this works on MacOS but not on Ubuntu 🤦‍♂️
Hmm I think back in the day this also worked for Ubuntu 😬

Comment (TGotwig) [reply Makefile:16]: Well, I think I will add a comment that this only works on MacOS currently, upload my MacOS archive on that formula repo and publish it that way 😊

**Comment by TGotwig** [comment]
Soo finished! 🚀
Can be installed on MacOS by:
```shell
brew tap tgotwig/sic
brew install tgotwig/sic/sic
```

Let me create another pull-request to mention this in the README.md 🙂

**Comment by foresterre** [comment]
Feel free to do that in this PR as well :)

**Comment by TGotwig** [comment]
That one extra commit regarding image optimizations comes from there:
https://github.com/TGotwig/sic/pull/1

**Comment by TGotwig** [comment]
Yeah I'm also not sure about that,
I removed the commit with the image optimizations 👍

**Review by foresterre — APPROVED** [review]

**Pull request merged**